Client/WarFare/GameProcMain.cpp

```diff
@@ -5756,38 +5758,57 @@ void CGameProcMain::ParseChattingCommand(const std::string& szCmd)
 		}
 		break;
 
-		// NOTE(Gilad): Can be better of course and using date time libraries, but I'm happy with this. feel free to change.
-		case CMD_TIME: 
+		case CMD_TIME:
 		{
 			std::string sTime = szCmds[1];
-			std::string sDate = szCmds[2];
-			int iYear  = 0;
-			int iMonth = 0;
-			int iDay   = 0;
-			int iHour  = 0;
-			int iMin   = 0;
+
+			// Commenting this out as we don't need to update date when using the /time command.
+			// But leaving the code below to support date too in case someone will need to use it in the future for whatever reason.
+			std::string sDate /*= szCmds[2]*/; 
+
+			int iYear = 0, iMonth = 0, iDay = 0, iHour = 0, iMin = 0;
+
+			tm tm = {};
+			std::istringstream i_sStream;
 
 			if (!sTime.empty())
 			{
-				size_t pos = sTime.find(":");
-				iHour = stoi(sTime.substr(0, pos));
-				sTime.erase(0, pos + 1);
-				iMin  = stoi(sTime);
-				if (iMin < 0 || iMin > 60) iMin = 0;
-			}
+				std::string sDateTime = sTime;
+				const char * c_pFormat = sTime.length() <= 2 ? "%H" : "%H:%M";
 
-			if (!sDate.empty())
-			{
-				size_t pos = sDate.find(".");
-				iYear = stoi(sDate.substr(0, pos));
-				sDate.erase(0, pos + 1);
-				pos   = sDate.find(".");
-				iMonth= stoi(sDate.substr(0, pos));
-				sDate.erase(0, pos + 1);
-				iDay  = stoi(sDate);
-
-				if (iMonth < 0 || iMonth > 12) iMonth = 1;
-				if (iDay   < 0 || iDay   > 31) iDay   = 1; // TODO(Gilad): Was too lazy to implement the day for each month logic..please forgive me.
+				if (!sDate.empty() && !sTime.empty())
+				{
+					sDateTime = sDate + " " + sTime;
+					c_pFormat = "%Y-%m-%d %H:%M";
+				}
+
+				i_sStream.str(sDateTime);
+				i_sStream >> std::get_time(&tm, c_pFormat);
+
+				if (!i_sStream.fail()) 
+				{
+					iHour = tm.tm_hour;
+					iMin  = tm.tm_min;
+					if (!sDate.empty())
+					{
+						iYear  = tm.tm_year + 1900;
+						iMonth = tm.tm_mon + 1;
+						iDay   = tm.tm_mday;
+
+						// Logic for max days in a month.
```

Comment:
All of the logic from here on is pointless; tm_mday -- and the other fields -- should always be valid when the structure's updated, which std::get_time() will be doing.